devicetypes/smartenit/smartenit-moisture-sensor.src/smartenit-moisture-sensor.groovy

```diff
@@ -0,0 +1,179 @@
+ */
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition(name: "Smartenit Moisture Sensor", namespace: "Smartenit", author: "Luis Contreras") {
+		capability "Battery"
+        capability "Configuration"
+		capability "Refresh"
+		capability "Water Sensor"
+		capability "Health Check"
+		capability "Sensor"
+
+		fingerprint manufacturer: "Compacta", model: "ZBLIQS", deviceJoinName: "Smartenit Moisture Sensor"
+	}
+}
+
+def getBATTERY_VOLTAGE_ATTR() { 0x0020 }
+
+def installed() {
+	log.debug "Installed"
+	createChildDevices()
+}
+
+private List<Map> collectAttributes(Map descMap) {
+	List<Map> descMaps = new ArrayList<Map>()
+
+	descMaps.add(descMap)
+
+	if (descMap.additionalAttrs) {
+		descMaps.addAll(descMap.additionalAttrs)
+	}
+
+	return  descMaps
+}
+
+def parse(String description) {
+	log.debug "description: $description"
+
+	// getEvent will handle temperature and humidity
+	Map map = zigbee.getEvent(description)
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			map = parseIasMessage(description)
+		} else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+
+			if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+            	map = getBatteryResult(Integer.parseInt(descMap.value, 16))
+            } else if (descMap?.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS) {
+				def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+				map = translateZoneStatus(zs)
+			}
+		}
+	}
+
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+
+	if (description?.startsWith('enroll request')) {
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+		result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+	}
+	return result
+}
+
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+
+	translateZoneStatus(zs)
+}
+
+private Map translateZoneStatus(ZoneStatus zs) {
+	if (zs.isAlarm2Set()) {
+		setChildMoistureState("wet")
+	} else {
+		setChildMoistureState("dry")
+	}
+	return zs.isAlarm1Set() ? getMoistureResult('wet') : getMoistureResult('dry')
+}
+
+private setChildMoistureState(value) {
+	def childDevice = childDevices.find {
+		it.deviceNetworkId == "$device.deviceNetworkId:02" || it.deviceNetworkId == "$device.deviceNetworkId:02"
+	}
+
+	if (childDevice) {
+		def map = createEvent(name: "water", value: value, translatable: true)
+		childDevice.sendEvent(map)
+	}
+}
+
+private Map getMoistureResult(value) {
+	log.debug "water"
+	def descriptionText
+	if (value == "wet")
+		descriptionText = '{{ device.displayName }} is wet'
+	else
+		descriptionText = '{{ device.displayName }} is dry'
+	return [
+		name           : 'water',
+		value          : value,
+		descriptionText: descriptionText,
+		translatable   : true
+	]
```

Comment:
Please fix the indentation (add 1 tab in every line)

Reply:
Hmm... I'm a bit confused as to how this should look. 😮 Am I adding a tab to lines 116-119 only?

Reply:
It's ok now, at first it looked like it was returned from previous `else` block, so please add brackets above -> 
`if () {...} else {...}`